Implement GC.GetGCMemoryInfo

This function exists in Dotnet 3.0, but is not yet implemented in Mono. The description in dotnet core is here:

https://docs.microsoft.com/en-us/dotnet/api/system.gcmemoryinfo?view=netcore-3.0

TODO:

- [ ] Reenable System.Tests.GCExtendedTests.GetGCMemoryInfo test in CoreFX.issues.rsp

New failure for reference:

```
FAILED: HighMemoryLoadThresholdBytes

Unhandled Exception:
Xunit.Sdk.EqualException: Assert.Equal() Failure
Expected: 17498344
Actual:   16777216
   at Xunit.Assert.Equal[Int64](Int64 expected, Int64 actual, IEqualityComparer`1 comparer)
   at Xunit.Assert.Equal[Int64](Int64 expected, Int64 actual)
   at System.Tests.GCExtendedTests.<>c.<GetGCMemoryInfo>b__22_0() in /_/src/System.Runtime/tests/System/GCTests.cs:line 818
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/ryan/git/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:line 339
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args)
[ERROR] FATAL UNHANDLED EXCEPTION: Xunit.Sdk.EqualException: Assert.Equal() Failure
Expected: 17498344
Actual:   16777216
   at Xunit.Assert.Equal[Int64](Int64 expected, Int64 actual, IEqualityComparer`1 comparer)
   at Xunit.Assert.Equal[Int64](Int64 expected, Int64 actual)
   at System.Tests.GCExtendedTests.<>c.<GetGCMemoryInfo>b__22_0() in /_/src/System.Runtime/tests/System/GCTests.cs:line 818
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) in /Users/ryan/git/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:line 339
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.DotNet.RemoteExecutor.Program.Main(String[] args)
    System.Tests.GCExtendedTests.GetGCMemoryInfo [FAIL]
      Microsoft.DotNet.RemoteExecutor.RemoteExecutionException : Remote process failed with an unhandled exception.
      Stack Trace:
        
        Child exception:
          Xunit.Sdk.EqualException: Assert.Equal() Failure
        Expected: 17498344
        Actual:   16777216
        /_/src/System.Runtime/tests/System/GCTests.cs(818,0): at System.Tests.GCExtendedTests.<>c.<GetGCMemoryInfo>b__22_0()
        /Users/ryan/git/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs(339,0): at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
        
        Child process:
          System.Runtime.Tests, Version=5.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51 System.Tests.GCExtendedTests+<>c Void <GetGCMemoryInfo>b__22_0()
```

--- a/netcore/CoreFX.issues.rsp
+++ b/netcore/CoreFX.issues.rsp
@@ -738,10 +738,6 @@
 ##  System.Runtime.Extensions.Tests
 ####################################################################
 
-# GC.GetGCMemoryInfo is not implemented
-# https://github.com/mono/mono/issues/15236
--nomethod System.Tests.AppDomainTests.MonitoringIsEnabled
-
 # Mono expands generic Types in stacktraces unlike the .NET Core (sounds more like a feature: https://gist.github.com/EgorBo/3abb37d2ff4fc904bc7472c62498f933)
 # https://github.com/mono/mono/issues/15315
 -nomethod System.Tests.EnvironmentStackTrace.StackTraceTest

